src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/DocumentBasedFixAllProvider.cs

```diff
@@ -30,8 +30,16 @@ namespace Microsoft.CodeAnalysis.CodeFixes
     /// </remarks>
     public abstract class DocumentBasedFixAllProvider : FixAllProvider
     {
+        private readonly ImmutableArray<FixAllScope> _supportedFixAllScopes;
+
         protected DocumentBasedFixAllProvider()
         {
+            _supportedFixAllScopes = base.GetSupportedFixAllScopes().ToImmutableArray();
```

Comment:
fwiw, this feels pretty wonky.  can we not hav FixAllProvider just make a protected internal static with the default `ImmutableArray.Create(FixAllScope.Document, FixAllScope.Project, FixAllScope.Solution)`?  the base woudl just return that.  here, we'd just defer to our other constructor, passing that along.

Reply:
Done.